src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -504,9 +504,13 @@ private TypeSymbol GetTypeOfTypeRef(TypeReferenceHandle typeRef, out bool isNoPi
 
             if (tokenType == HandleKind.AssemblyReference)
             {
-                // TODO: Can refer to the containing assembly?
                 isNoPiaLocalType = false;
-                return LookupTopLevelTypeDefSymbol(Module.GetAssemblyReferenceIndexOrThrow((AssemblyReferenceHandle)tokenResolutionScope), ref fullName);
+                var assemblyRef = (AssemblyReferenceHandle)tokenResolutionScope;
```

Comment:
We should throw BadImage for nil ModuleRef and TypeRef as well. 

If the token is 0x00000000 (IsNil == true, tokenType == ModuleDefinition) we should return ```GetUnsupportedMetadataTypeSymbol();```. The spec allows null token:

> null - in this case, there shall be a row in the ExportedType table for this Type
> - its Implementation field shall contain a File token or an AssemblyRef token
> that says where the type is defined [ERROR]

We just don't support it.